### PR TITLE
#18 add route api/user/bids

### PR DESCRIPTION
**Pull request by bnonni**
WIP PR - actively developing for https://github.com/PlebFM/PlebFM/issues/18

### CODE REVIEW

**Comment by bnonni** [comment]
No longer WIP. Please review at leisure @alexlwn123

File: pages/api/bid/history.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        return res
+            .status(200)
+            .json({ success: true, new: false, message: bids });
+    } catch (error: any) {
+        return res.status(500).json({ success: false, message: error.message });
+    }
+};
+
+export default connectDB(handler);
```
Comment (alexlwn123) [inline pages/api/bid/history.ts:35]: Did the prettier/husky script not run automatically? Looks like the formatting didn't get fixed 😢

**Review by alexlwn123 — CHANGES_REQUESTED** [review]
Now that `main` is stable (builds successfully), I added the branch protections back in. Just pull in main and double check that `npm build` succeeds.

Other than that LGTM! Thanks!

File: pages/api/bid/history.ts (hunk 1 above)
Comment (bnonni) [reply pages/api/bid/history.ts:33]: It didn't, but after checking out main, running npm install and then running npm run format, im all good. latest push should be good to go.

**Review by bnonni — COMMENTED** [review]
Formatting fixed (i think)

**Comment by bnonni** [comment]
`npm run build` is throwing errors 

```
./components/AlbumScreen.tsx:9:3
Type error: Member 'track' implicitly has an 'any' type.

   7 | import NavBar from './NavBar';
   8 | export default function AlbumScreen(props: {
>  9 |   track;
     |   ^
  10 |   setSongConfirmed;
  11 |   cancelSong;
  12 | }) {

./components/PaymentScreen.tsx:10:3
Type error: Member 'song' implicitly has an 'any' type.

   8 | 
   9 | function PaymentScreen(props: {
> 10 |   song;
     |   ^
  11 |   readyToCheckout;
  12 |   invoicePaid;
  13 |   setInvoicePaid;
```

**Review by alexlwn123 — APPROVED** [review]
Looks great. Thanks!

**Pull request merged**